### PR TITLE
ci(backends): run backend doctests in CI

### PR DESCRIPTION
**Pull request by cpcloud**
Run backend doctests in CI during the usual backend test suite.

### CODE REVIEW

File: ibis/backends/postgres/__init__.py [hunk 1]
```diff
@@ -228,8 +228,8 @@ def do_connect(
         >>> import getpass
```
Comment (gmcrocetti) [inline ibis/backends/postgres/__init__.py:228]:
```suggestion

```

**Comment by gmcrocetti** [comment]
👏 . LMK in case you need/want help breaking up into smaller tasks o/.

File: ibis/backends/postgres/__init__.py [hunk 2]
```diff
@@ -225,11 +225,10 @@ def do_connect(
         Examples
         --------
         >>> import os
-        >>> import getpass
         >>> import ibis
         >>> host = os.environ.get("IBIS_TEST_POSTGRES_HOST", "localhost")
-        >>> user = os.environ.get("IBIS_TEST_POSTGRES_USER", getpass.getuser())
-        >>> password = os.environ.get("IBIS_TEST_POSTGRES_PASSWORD")
+        >>> user = os.environ.get("IBIS_TEST_POSTGRES_USER", "postgres")
```
Comment (gmcrocetti) [inline ibis/backends/postgres/__init__.py:230]: This is me being a nitpicker but do you believe we can replace `IBIS_TEST_*` prefixes by their common postgres version ?
- IBIS_TEST_POSTGRES_HOST => PGHOST
- IBIS_TEST_POSTGRES_USER => PGUSER
- IBIS_TEST_POSTGRES_PASSWORD =>  PGPASSWORD
- IBIS_TEST_POSTGRES_DATABASE =>  PGDATABASE
Saw this in other places as well maybe I can replace it all ?

Comment (cpcloud) [reply ibis/backends/postgres/__init__.py:213]: I think so, but let's do that in a follow up!

**Comment by gforsyth** [comment]
Well, I'd say the doctests in the backend were a good idea, even if it's only turning up out-of-date reprs

**Comment by cpcloud** [comment]
@gforsyth Indeed!

Thanks @gmcrocetti for pushing on this one!

**Review by gforsyth — APPROVED** [review]
good to go if CI is green

Thanks for leading the way here @gmcrocetti !

**Comment by cpcloud** [comment]
Lemme make sure clouds are clouding.

**Comment by cpcloud** [comment]
Not much going on in the way of ☁️ doctests, but all good nonetheless:

```
❯ just backend-doctests snowflake -q && just backend-doctests bigquery -q
..                                                                                                                                                                                                             [100%]
2 passed in 0.44s
.                                                                                                                                                                                                              [100%]
1 passed in 1.12s
```

**Pull request merged**